spec/lib/cache_spec.rb

```diff
@@ -121,5 +121,25 @@
         expect(cache.read("my_key")).to eq("bob")
       end
     end
+
+    it "works even when the key expires during fetch" do
+      original_get = Discourse.redis.method(:get)
+
+      calls = 0
+
+      # 1st call returns the value
+      # further calls return nil to simulate the key expiring
+      Discourse
+        .redis
+        .define_singleton_method(:get) do |*args|
+          calls += 1
+          calls == 1 ? Marshal.dump("bob") : nil
+        end
+
+      expect(fetch_value).to eq("bob")
+      expect(calls).to eq(1)
+
+      Discourse.redis.define_singleton_method(:get, original_get)
```

Comment:
If we're keeping this, then it needs to go in an `ensure` block. Otherwise, if this spec fails, it will never be reset, and it will pollute the rest of the test suite.